apis/capabilities/v1beta1/product_types.go

```diff
@@ -322,14 +322,86 @@ func (u *UserKeyAuthenticationSpec) GatewayResponseSpec() *GatewayResponseSpec {
 	return u.GatewayResponse
 }
 
+// OIDCAuthenticationFlowSpec defines the desired OAuth2.0 authorization grant type
+type OIDCAuthenticationFlowSpec struct {
```

Comment:
Are they all optional? if so we should set them as pointers and mark them as optional appropriately as per k8s API conventions guide (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md)

Reply:
They are all required. It is the attribute of the OIDCSpec `AuthenticationFlow` the one which is optional.
```
// AuthenticationFlow specifies OAuth2.0 authorization grant type                              
// +optional                                                                                   
AuthenticationFlow *OIDCAuthenticationFlowSpec `json:"authenticationFlow,omitempty"`           
```

Reply:
:+1: